source/command-logging-and-monitoring/tests/monitoring/writeConcernError.yml

```diff
@@ -0,0 +1,80 @@
+description: "writeConcernError"
+schemaVersion: "1.13"
+runOnRequirements:
+  -
+    minServerVersion: 4.1.0
+    topologies:
+      - replicaset
+    serverless: "forbid"
+
+createEntities:
+  - client:
+      id: &client client
+      observeEvents:
+        - commandStartedEvent
+        - commandSucceededEvent
+        - commandFailedEvent
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName command-monitoring-tests
+  - collection:
+      id: &collection collection
+      database: *database
+      collectionName: &collectionName test
+
+initialData:
+  - collectionName: *collectionName
+    databaseName: *databaseName
+    documents:
+      - { _id: 1, x: 11 }
+
+tests:
+  - description: "A retryable write with write concern errors publishes success event"
+    operations:
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *client
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+              failCommands: [ insert ]
+              writeConcernError:
+                code: 91 # ShutdownInProgress
+                errorLabels: [RetryableWriteError]
+      - name: insertOne
```

Comment:
is the extra operation and started/succeeded event necessary?

Reply:
I wanted to do something similar to the [retryable writes prose test #3](https://github.com/mongodb/specifications/tree/master/source/retryable-writes/tests#prose-tests) where the first write concern error causes a retry. So in this case we get the extra started/succeeded event because it does retry the first insert.